projects/packages/paypal-payments/src/legacy/class-simple-payments.php

```diff
@@ -0,0 +1,834 @@
+		add_action( 'rest_api_init', array( $this, 'register_meta_fields_in_rest_api' ) );
+	}
+
+	/**
+	 * Register the shortcode.
+	 */
+	private function register_shortcode() {
+		add_shortcode( self::$shortcode, array( $this, 'parse_shortcode' ) );
+	}
+
+	/**
+	 * Actions that are run on init.
+	 */
+	public function init_hook_action() {
+		add_filter( 'rest_api_allowed_post_types', array( $this, 'allow_rest_api_types' ) );
+		add_filter( 'jetpack_sync_post_meta_whitelist', array( $this, 'allow_sync_post_meta' ) );
+		if ( ! is_admin() ) {
+			$this->register_scripts_and_styles();
+		}
+		$this->register_shortcode();
+		$this->setup_cpts();
+
+		add_filter( 'the_content', array( $this, 'remove_auto_paragraph_from_product_description' ), 0 );
+	}
+
+	/**
+	 * Enqueue the static assets needed in the frontend.
+	 */
+	public function enqueue_frontend_assets() {
+		if ( ! wp_style_is( 'jetpack-simple-payments', 'enqueued' ) ) {
+			wp_enqueue_style( 'jetpack-simple-payments' );
+		}
+
+		if ( ! wp_script_is( 'jetpack-paypal-express-checkout', 'enqueued' ) ) {
+			wp_enqueue_script( 'jetpack-paypal-express-checkout' );
+		}
+	}
+
+	/**
+	 * Add an inline script for setting up the PayPal checkout button.
+	 *
+	 * @param int     $id Product ID.
+	 * @param int     $dom_id ID of the DOM element with the purchase message.
+	 * @param boolean $is_multiple Whether multiple items of the same product can be purchased.
+	 */
+	public function setup_paypal_checkout_button( $id, $dom_id, $is_multiple ) {
+		wp_add_inline_script(
+			'jetpack-paypal-express-checkout',
+			sprintf(
+				"try{PaypalExpressCheckout.renderButton( '%d', '%d', '%s', '%d' );}catch(e){}",
+				esc_js( $this->get_blog_id() ),
+				esc_js( $id ),
+				esc_js( $dom_id ),
+				esc_js( $is_multiple )
+			)
+		);
+	}
+
+	/**
+	 * Remove auto paragraph from product description.
+	 *
+	 * @param string $content - the content of the post.
+	 */
+	public function remove_auto_paragraph_from_product_description( $content ) {
+		if ( get_post_type() === self::$post_type_product ) {
+			remove_filter( 'the_content', 'wpautop' );
+		}
+
+		return $content;
+	}
+
+	/** Return the blog ID */
+	public function get_blog_id() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			return get_current_blog_id();
+		}
+
+		return Jetpack_Options::get_option( 'id' );
+	}
+
+	/**
+	 * Used to check whether Simple Payments are enabled for given site.
+	 *
+	 * @return bool True if Simple Payments are enabled, false otherwise.
+	 */
+	public function is_enabled_jetpack_simple_payments() {
+		/**
+		 * Can be used by plugin authors to disable the conflicting output of Simple Payments.
+		 *
+		 * @since 6.3.0
+		 *
+		 * @param bool True if Simple Payments should be disabled, false otherwise.
+		 */
+		if ( apply_filters( 'jetpack_disable_simple_payments', false ) ) {
+			return false;
+		}
+
+		return ( ( defined( 'IS_WPCOM' ) && IS_WPCOM )
+			|| Jetpack::is_connection_ready() )
```

Comment:
You should be able to switch to relying on the Connection package for this (and thus require the connection package in the PayPal Payments package).

Reply:
Thanks for flagging this and sharing the Connection package. I've added the package as a dependency in [1632a85](https://github.com/Automattic/jetpack/pull/43413/commits/1632a85f6ded84bcff4834646532227eb6a77729).

I've replaced the method call in [e784072](https://github.com/Automattic/jetpack/pull/43413/commits/e78407258634a0d92750324ad015a7eee5999b46).